pkg/epp/datastore/datastore.go

```diff
@@ -116,6 +117,14 @@ func (ds *datastore) PoolSet(ctx context.Context, reader client.Reader, pool *v1
 
 	oldPool := ds.pool
 	ds.pool = pool
+	if oldPool == nil || pool.Spec.TargetPorts[0] != oldPool.Spec.TargetPorts[0] {
+		if datalayer.Enabled(logger) {
+			if source, found := datalayer.GetNamedSource(dlmetrics.DataSourceName); found {
+				metricsSource := source.(*dlmetrics.DataSource)
+				metricsSource.SetPort(int32(pool.Spec.TargetPorts[0].Number))
+			}
+		}
+	}
```

Comment:
this is not always true. if the metrics port was set from the cmd-lind arguments, this check may cause a bug.